plugins/pipette/src/pipette.cc

```diff
@@ -0,0 +1,223 @@
+#include "pipette.hh"
+
+#include <gdk/gdk.h>
+#include <cmath>
+
+#include <scroom/pipetteviewinterface.hh>
+#include <scroom/unused.hh>
+
+////////////////////////////////////////////////////////////////////////
+// Pipette
+////////////////////////////////////////////////////////////////////////
+
+Pipette::Pipette()
+{
+}
+Pipette::~Pipette()
+{
+}
+
+Pipette::Ptr Pipette::create()
+{
+  return Ptr(new Pipette());
+}
+
+////////////////////////////////////////////////////////////////////////
+// PluginInformationInterface
+////////////////////////////////////////////////////////////////////////
+
+std::string Pipette::getPluginName()
+{
+  return "Pipette";
+}
+
+std::string Pipette::getPluginVersion()
+{
+  return "0.0";
+}
+
+void Pipette::registerCapabilities(ScroomPluginInterface::Ptr host)
+{
+  host->registerViewObserver("Pipette", shared_from_this<Pipette>());
+}
+
+////////////////////////////////////////////////////////////////////////
+// ViewObserver
+////////////////////////////////////////////////////////////////////////
+
+Scroom::Bookkeeping::Token Pipette::viewAdded(ViewInterface::Ptr view)
+{
+  PipetteHandler::Ptr handler = PipetteHandler::create();
+  view->registerSelectionListener(handler);
+  view->registerPostRenderer(handler);
+
+  view->addToolButton(GTK_TOGGLE_BUTTON(gtk_toggle_button_new_with_label("Pipette")), handler);
+
+  return Scroom::Bookkeeping::Token();
+}
+
+////////////////////////////////////////////////////////////////////////
+// PipetteHandler
+////////////////////////////////////////////////////////////////////////
+
+PipetteHandler::PipetteHandler()
+{
+  selection = nullptr;
+  enabled = false;
+  pendingJobs = ThreadPool::Queue::createAsync();
+}
+PipetteHandler::~PipetteHandler()
+{
+}
+
+PipetteHandler::Ptr PipetteHandler::create()
+{
+  return Ptr(new PipetteHandler());
+}
+
+void PipetteHandler::computeValues(ViewInterface::Ptr view)
+{
+  gdk_threads_enter();
+  view->setStatusMessage("Computing colors values...");
```

Comment:
I think "color" should be singular here.

Reply:
Yes, you're correct